### PR TITLE
Effective AoA Aft Feature: thin-airfoil downwash correction for tandem

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

The aft foil in a tandem configuration operates in the wake of the fore foil. The fore foil's circulation creates a **downwash** that shifts the effective angle-of-attack seen by the aft foil. Classical thin-airfoil theory gives: `AoA_effective ≈ AoA_global + k * AoA_global / gap_normalized`, where k is a coupling coefficient.

This aerodynamic interference effect is **entirely absent from the current input space** — the model only sees global AoA, not the modified effective AoA at the aft foil. For single-foil samples, effective AoA = global AoA (no correction). For tandem, the correction is significant, especially at large AoA / small gap.

**Why this should work:** Every p_tan win has come from explicit inter-foil geometry/physics features. The wake deficit feature (PR #2213, -4.1% p_in) encoded geometric displacement. This encodes **aerodynamic interference** (AoA perturbation). They target different aspects of fore→aft coupling and are complementary.

## Instructions

Add 1 new input channel: effective AoA for the aft foil, computed from thin-airfoil downwash theory. Sweep over coupling coefficient k={0.05, 0.10, 0.20}.

### Implementation

1. **Identify tandem vs single-foil samples** in the batch. The data pipeline should have a flag or you can infer from gap/stagger being nonzero.

2. **Compute effective AoA:**
```python
# Global scalars already in input features:
aoa_val = x[:, 0, aoa_channel]   # [B] — global AoA in radians
gap_norm = x[:, 0, gap_channel]  # [B] — normalized gap (already in features)

k = 0.10  # coupling coefficient (hyperparameter)

# For tandem: effective AoA = AoA + k * AoA / (|gap| + epsilon)
# For single-foil: effective AoA = AoA (no correction)
is_tandem = (gap_norm.abs() > 1e-4)  # [B] bool

eff_aoa = torch.where(
    is_tandem,
    aoa_val + k * aoa_val / (gap_norm.abs() + 0.1),  # downwash correction
    aoa_val  # single-foil: unchanged
)

# Broadcast to all nodes as new input channel
eff_aoa_feature = eff_aoa.unsqueeze(-1).unsqueeze(-1).expand(-1, N, -1)  # [B, N, 1]
x = torch.cat([x, eff_aoa_feature], dim=-1)  # [B, N, 25]
```

3. **Add flag** `--effective_aoa_feature` and `--effective_aoa_k 0.10` (float). Update `input_dim` calculation to add 1 when this flag is set.

4. **Run 3 values of k** using `--wandb_group "effective-aoa-aft"`:
   - k=0.05 (weak coupling)
   - k=0.10 (moderate coupling) — **start here**
   - k=0.20 (strong coupling)

5. **Important:** Read `prepare_multi.py` to find the correct channel indices for AoA and gap. The gap might be encoded as normalized gap-stagger features.

### Training command (seed 42, k=0.10)
```bash
cd cfd_tandemfoil && python train.py --agent tanjiro --seed 42 \
  --wandb_name "tanjiro/effective-aoa-k0.10-s42" \
  --wandb_group "effective-aoa-aft" \
  --effective_aoa_feature --effective_aoa_k 0.10 \
  --asinh_pressure --asinh_scale 0.75 --field_decoder --adaln_output --use_lion --lr 2e-4 \
  --aug aoa_perturb --aug_full_dsdf_rot --high_p_clamp --n_layers 3 --slice_num 96 \
  --tandem_ramp --domain_layernorm --domain_velhead --ema_decay 0.999 --weight_decay 5e-5 \
  --cosine_T_max 150 --pcgrad_3way --pcgrad_extreme_pct 0.15 \
  --pressure_first --pressure_deep \
  --residual_prediction --surface_refine --surface_refine_hidden 192 --surface_refine_layers 3 \
  --aft_foil_srf --aug_gap_stagger_sigma 0.02 --aug_dsdf2_sigma 0.05 \
  --gap_stagger_spatial_bias \
  --dct_freq_loss --dct_freq_weight 0.05 --dct_freq_gamma 2.0 --dct_freq_alpha 1.5 \
  --te_coord_frame --wake_deficit_feature
```

Seed 73: same with `--seed 73` and adjusted `--wandb_name`.

**If k=0.10 looks promising, run k=0.05 and k=0.20 as follow-ups.** If k=0.10 clearly fails, report and close — no need for the sweep.

### What to report
1. 2-seed average for p_in, p_oodc, p_tan, p_re (for each k value tested)
2. W&B run IDs
3. Whether the feature has different effects on tandem vs single-foil validation splits

## Baseline (PR #2251, 2-seed avg)

| Metric | 2-seed avg | Target to beat |
|--------|-----------|----------------|
| **p_in** | **11.891** | < 11.89 |
| **p_oodc** | **7.561** | < 7.56 |
| **p_tan** | **28.118** | < 28.12 |
| p_re | 6.364 | < 6.36 |

W&B baseline: 7jix2jkg (s42), epkfhxfl (s73)

### CODE REVIEW

**Comment by tcapelle** [comment]
## Experiment Results: Effective AoA Aft Feature (k=0.10)

### Setup
- **Hypothesis**: Thin-airfoil downwash correction — `AoA_eff = AoA + k * AoA / (|gap| + 0.1)` for tandem, unchanged for single-foil
- **Flags**: `--effective_aoa_feature --effective_aoa_k 0.10`
- **Added 1 new input channel** appended after wake deficit features, before Fourier PE

### Results (surface MAE pressure)

| Metric | Baseline (PR #2251) | Seed 42 | Seed 73 | Avg |
|--------|-------------------|---------|---------|-----|
| p_in   | **11.891** | 11.9 ❌ | 13.1 ❌ | 12.5 |
| p_oodc | **7.561** | 7.9 ❌ | 7.9 ❌ | 7.9 |
| p_tan  | **28.118** | 29.1 ❌ | 28.9 ❌ | 29.0 |
| p_re   | **6.364** | 6.5 ❌ | 6.5 ❌ | 6.5 |

### W&B Runs
- Seed 42: `42otg68a` — [view run](https://wandb.ai/wandb-applied-ai-team/senpai-v1/runs/42otg68a)
- Seed 73: `1fxgd7h3` — [view run](https://wandb.ai/wandb-applied-ai-team/senpai-v1/runs/1fxgd7h3)
- Group: `effective-aoa-aft`

### Analysis
- **All metrics worse than baseline across both seeds.** The feature hurts rather than helps.
- **k=0.05 and k=0.20 sweep not run** — k=0.10 clearly failed, so no point in sweeping.
- The thin-airfoil downwash model may be too simplified for the actual wake interactions in this dataset. The wake deficit feature (PR #2213) likely captures the relevant inter-foil coupling more accurately through geometric displacement rather than a linearized AoA correction.
- Seed 73 p_in is notably worse (13.1 vs 11.9), suggesting the feature may inject noise into single-foil predictions too, despite the tandem-only gating.
- Note: flow field visualization failed in both runs with a dimension mismatch (`66 vs 67`), suggesting the viz path may need the effective_aoa_feature too — but this doesn't affect training or metrics.

### Suggested Follow-ups
- The linearized thin-airfoil model is too crude for this dataset. If revisiting inter-foil AoA coupling, consider a learned correction rather than a fixed formula.
- The existing wake_deficit_feature may already capture what this was trying to model, making them redundant.

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2287 because ADVISOR: Closing — effective AoA aft feature regresses all metrics (p_in +5.1%, p_oodc +4.5%, p_tan +3.1%, p_re +2.1%).

**Root cause — thin-airfoil downwash model too simplified.** The linearized formula AoA + k*AoA/gap is a very crude approximation of actual wake interactions. The existing wake_deficit_feature (PR #2213) already captures inter-foil coupling more accurately through geometric displacement (gap-normalized TE offset) rather than a linearized AoA correction. The two features are likely redundant, with wake_deficit being strictly superior.

Good call on skipping the k sweep given the clear negative at k=0.10. Moving you to a fresh direction..
